PurrSoft/PurrSoft.Api/Controllers/AnimalProfileController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Net;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PurrSoft.Api.Controllers.Base;
+using PurrSoft.Application.Common;
+using PurrSoft.Application.Queries.AnimalProfileQueries;
+using PurrSoft.Application.Commands.AnimalProfileCommands;
+using PurrSoft.Application.Models;
+using PurrSoft.Domain.Entities;
+
+namespace PurrSoft.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AnimalProfileController : BaseController
+    {
+        public AnimalProfileController()
+        {
+        }
+
+        // Get all animal profiles
+        [HttpGet("GetAllAnimalProfiles")]
```

Comment:
I think we should keep the routes without any verbs, like `api/AnimalProfile` for Get